packages/router/utils.ts

```diff
@@ -1047,15 +1047,15 @@ export interface TrackedPromise extends Promise<any> {
 
 export class AbortedDeferredError extends Error {}
 
-export class DeferredData {
+export class DeferredData<Data = Record<string, unknown>> {
```

Comment:
It's a best practice to also declare what kind of type it should extend

```suggestion
export class DeferredData<Data extends Record<string, unknown> = Record<string, unknown>> {
```